src/codeu/chat/client/commandline/Chat.java

```diff
@@ -400,96 +442,65 @@ public void invoke(List<String> args) {
-        }
-        System.out.println("--- update: conversation interests ---");
-        for (final ConversationInterestContext interest : user.conversationInterests()) {
-          System.out.format("CONVERSATION : %s \nNumber of new messages : %s\n",
-                  interest.conversation.title,
-                  interest.interest.messageCount);
+      private ConversationContext findConversation(String title) {
+        for (final ConversationContext other : user.conversations()) {
+          if (title.equals(other.conversation.title)) {
+            return other;
+          }
         }
+        return null;
       }
     });
 
-    /*
-    // i-convo-update (update a conversation interest)
+    // i-status-update (status update for a user's interests)
     //
     // Add a command to add a new interest when the user enters
     // convI-add while on the interest panel.
     //
-    panel.register("i-convo-update", new Panel.Command() {
+    panel.register("i-status-update", new Panel.Command() {
       @Override
       public void invoke(List<String> args) {
-        final String title = args.size() > 0 ? args.get(0) : "";
-        if (title.length() > 0) {
-          final ConversationInterestContext conversationInterest = findConversationInterest(title);
-          final ConversationContext conversation = findConversation(title);
-
-          if (conversationInterest != null && conversation != null) {
-              System.out.println("--- start of conversation ---");
-              for (MessageContext message = conversation.firstMessage();
-                   message != null;
-                   message = message.next()) {
-                System.out.println();
-                System.out.format("USER : %s\n", message.message.author);
-                System.out.format("SENT : %s\n", message.message.creation);
-                System.out.println();
-                System.out.println(message.message.content);
-                System.out.println();
-              }
-              System.out.println("Number of missed Messages: " + conversationInterest.interest.messageCount);
-              System.out.println("---  end of conversation  ---");
-            } else {
-            System.out.format("ERROR: No conversation or no interest found");
-          }
-        } else {
-          System.out.println("ERROR: Missing conversation <title>");
-        }
-      }
 
-      private ConversationContext findConversation(String title) {
-        for (final ConversationContext conversation : user.conversations()) {
-          if (title.equals(conversation.conversation.title)) {
-            return conversation;
+        Iterator<Interest> iterator = interestMap.get(user.user.id).iterator();
+        String userInterests = "";
+        String conversationInterests = "";
+        while(iterator.hasNext()){
+          Interest current = iterator.next();
+
+          if (current.getClass() == ConversationInterest.class) {
```

Comment:
I check the classes so I know which method to call to update the interest, if that makes sense